### PR TITLE
WSSQL-157 Platform as submodule

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@rpastrana 

- Added HPCC-Platform submodule to wssql project, with submodule set to candidate-6.4.0
- Cleaned up the documentation for building wssql
- Updated the root CMakeLists.txt file to use 3.2.2 by default for all platforms
- Updated the root CMakeLists.txt file to use consistent syntax/style

**Comment by Michael-Gardner** [comment]
@xwang2713 Can you please review this?  I'm thinking if you and Richard both approve of this method, we can extend it to other projects that require parallel builds.

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,176 +13,164 @@
...
 #           This file is the top level handling for
 #           cmake based compilation and build process.
 #
 #   To build for Linux:
-#   1. Check out source hpcc-systems/wssql
-#   2. Check out pre-req source hpcc-systems/HPCC-Platform (check ~/wssql/platform-version-prereq.cmake for minimum version required)
-#   3. Setup Antlr and Antlr3c.
-#    a. "antlr-${ANTLR_VER}-complete.jar" and "antlr-runtime-${ANTLR_VER}.jar" are required.
-#    b. Build process looks for these files here: /usr/local/ANTLR/${ANTLR_VER} where ANTLR_VER = 3.4
+#   1.  Check out source hpcc-systems/wssql
```
Comment (rpastrana) [inline CMakeLists.txt]: should the submodule update command be issued here?

File: CMakeLists.txt [hunk 2]
```diff
@@ -13,176 +13,164 @@
...
+set(CMAKE_PROJECT_NAME "hpccsystems-wssql")
 
-set (HPCC_WSSQL_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
-set (HPCC_SOURCE_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/..)
+set(HPCC_WSSQL_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 
-IF ("${HPCC_SOURCE_DIR}" STREQUAL "")
-  SET (HPCC_SOURCE_DIR ${HPCC_SOURCE_ROOT_DIR}/HPCC-Platform)
-ENDIF()
+if("${HPCC_SOURCE_DIR}" STREQUAL "")
```
Comment (rpastrana) [inline CMakeLists.txt]: can the build process be directed to use a local HPCC-Platform clone instead of the submodule?

File: CMakeLists.txt [hunk 3]
```diff
@@ -13,176 +13,164 @@
...
-                            ERROR_VARIABLE  packageRevision
-                    )
-
-        MESSAGE ( "-- Auto Detecting Packaging type")
-        MESSAGE ( "-- distro uses ${packageManagement}, revision is ${packageRevisionArch}" )
-
-        IF ( "${packageManagement}" STREQUAL "DEB" )
+    endif()
+
+    if(UNIX)
```
Comment (rpastrana) [inline CMakeLists.txt]: what happens if APPLE?

File: platform-version-prereq.cmake [hunk 4]
```diff
@@ -1,22 +0,0 @@
-###
```
Comment (rpastrana) [inline platform-version-prereq.cmake]: why is this file removed?

File: CMakeLists.txt (hunk 3 above)
Comment (Michael-Gardner) [inline CMakeLists.txt]: APPLE is a subset of UNIX.  If APPLE is true, UNIX will be true.

File: CMakeLists.txt (hunk 2 above)
Comment (Michael-Gardner) [inline CMakeLists.txt]: Yes.  If you choose to set HPCC_SOURCE_DIR to a different absolute path, then the build will use that instead of the submodule we default.

File: platform-version-prereq.cmake (hunk 4 above)
Comment (Michael-Gardner) [inline platform-version-prereq.cmake]: with a submodule build, we control the git reference that our submodule points too.  So having a platform version prereq file is unnecessary.

File: CMakeLists.txt (hunk 2 above)
Comment (rpastrana) [inline CMakeLists.txt]: Then let's document that in the README

File: platform-version-prereq.cmake (hunk 4 above)
Comment (rpastrana) [inline platform-version-prereq.cmake]: But the user is still able to point to an independent HPCC-Platform clone via HPCC_SOURCE_DIR.

File: CMakeLists.txt (hunk 1 above)
Comment (Michael-Gardner) [inline CMakeLists.txt]: Yes I can add that to the documentation

**Comment by Michael-Gardner** [comment]
@rpastrana I updated the sections you commented on.

**Comment by xwang2713** [comment]
OK
